packages/http/server/test/response_builder_spec.mare

```diff
@@ -0,0 +1,66 @@
+:import "spec"
+
+:class ResponseBuilderSpec
+  :is Spec
+  :const describes: "ResponseBuilder"
+
+  :it "builds 1xx responses"
+    builder = ResponseBuilder.new(Bytes.new)
+    builder.status_continue
+
+    // If we wanted to use `@assert = builder.finish == b"..."
+    // we would nead to teach ByteStreamWritable how to use the `==` function:
+    //
+    // :fun "=="(other Bytes'box) Bool: @ == other
+    //
+    // Instead I'm stealing this try block form the RequestParser spec.
+    @assert = try (
+      builder.finish.as!(Bytes'box) == b"HTTP/1.1 100 Continue\r\n\r\n"
+    |
+      False
+    )
+
+    builder = ResponseBuilder.new(Bytes.new)
+    builder.status_switching_protocols
+
+    @assert = try (
+      builder.finish.as!(Bytes'box) == b"HTTP/1.1 101 Switching Protocols\r\n\r\n"
+    |
+      False
+    )
+
+  // TODO: finish these test cases.
+  :it "builds 2xx responses"
+  :it "builds 3xx responses"
+  :it "builds 4xx responses"
+  :it "builds 5xx responses"
+
+  :it "can add headers"
+    builder = ResponseBuilder.new(Bytes.new)
+    builder.status_ok
+
+    builder.header("Content-Length", "0")
+
+    @assert = try (
+      builder.finish.as!(Bytes'box) == b"HTTP/1.1 200 OK\r\nContent-Length: 0\r\n\r\n"
+    |
+      False
+    )
+
+  :it "can write a body"
+    builder = ResponseBuilder.new(Bytes.new)
+    builder.status_ok
+
+    builder.header("Content-Type", "text/plain; charset=utf-8")
+    builder.write("Hello, World!")
+
+    // TODO: Writing the expected response like makes it is hard to read.
+    // `Array#join` would help, but doesn't exist. Maybe try some of the tricks
```

Comment:
Do we want a ticket for `Array#join`?

Reply:
I think `Array#join` from Ruby is a bit strange in a strongly typed language since it could be an array of anything.

Maybe something like `Bytes.join` and/or `String.join` would be good?

(recall that we can define a `:fun non join` on a type that acts like a `def self.join` from Ruby)

```mare
Bytes.join([
  b"HTTP/1.1 200 OK\r\n"
  b"Content-Type: text/plain; charset=utf-8\r\n"
  b"\r\n"
  b"Hello, World!\r\n"
])
```

Or perhaps with the `\r\n` delimiter separately specified:


```mare
Bytes.join([
  b"HTTP/1.1 200 OK"
  b"Content-Type: text/plain; charset=utf-8"
  b""
  b"Hello, World!"
  b""
], b"\r\n")
```

Reply:
I like this! And I _think_ Elixir takes a similar approach.

Do these functions exist? If not I'll either add it in the PR or create a new ticket.

Reply:
No, they don't exist yet - feel free to add them or create a ticket.